docs/source/design/user_messaging.rst

```diff
@@ -0,0 +1,124 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_user_messaging:
+
+*******************************************************
+User messaging: result records vs exceptions vs logging
+*******************************************************
+
+.. topic:: Specification scope and status
+
+   This specification aims to delineate the applicable contexts for using result records,
+   exceptions, logging, or other types of user messaging methods.
+
+Motivation
+==========
+
+Design specifications exist for :ref:`chap_design_result_records`,
+:ref:`chap_design_exception_handling`, :ref:`chap_design_log_levels`, and 
+:ref:`chap_design_progress_reporting`, and an ideal yet elusive goal is the consistent
+application of these methods in the applicable contexts. 
+
+From a **developer's perspective**, a specification for the proper use of specific user
+messaging methods creates a best practice standard that simplifies the implementation
+process, minimizes code duplication, and ensures consistency in the code base. This
+applies both to development in DataLad core as well as in DataLad extensions.
+
+From a **user's perspective**, it is imperative to receive the most appropriate and
+unambiguous message relating to the current context or action.
+
+Discrepancies (on the development side) and resulting ambiguity (on the user side) arise
+when the chosen user messaging methods are implemented interchangeably, inconsistently,
+within contexts that do not warrant them, and in ways that do not inform users
+appropriately.
+
+Consequently, the motivation of this specification is to delineate the applicable
+contexts for using result records, exceptions, logging, or other types of
+user messaging processes.
+
+
+Specification
+=============
+
+Result records
+--------------
+
+**Result records are the standard and preferred return value** format for all
+DataLad commands. Result records are routinely inspected throughout the code base,
+and are used to inform generic error handling, as well as particular calling commands
+on how to proceed with a specific operation.
+
+Yield result records when:
```

Comment:
Since seems to concentrate on "errors", is the intended meaning

```suggestion
Yield result records instead of raising an exception when:
```
?

Reply:
Restructured the section in [721c3d9](https://github.com/datalad/datalad/pull/7310/commits/721c3d9bce724a35a3235edecb7e8815c584d7fc), and perhaps this warrants a reassessment of these comments?